src/sage/graphs/matching_covered_graph.py

```diff
@@ -2033,12 +2033,9 @@ def maximal_barrier(self, vertex):
         if vertex not in self:
             raise ValueError('vertex {} not in the graph'.format(vertex))
 
-        G = Graph(self, multiedges=False)
-        M = Graph(self.get_matching())
-        B = set([vertex])
-
         # u: The M neighbor of vertex
-        u = next(M.neighbor_iterator(vertex))
+        matching = self.get_matching()
+        u = next((a if b == vertex else b) for a, b, *_ in matching if vertex in [a, b])
```

Comment:
why using `*_` ?  isn't `_` sufficient ?

Reply:
We might have an edge with just two parameters without any labels. In such a case, we might a following `ValueError` if we use `for a, b, _ in matching`:

```
Traceback (most recent call last)
...
ValueError: not enough values to unpack (expected 3, got 2)
```

And the following one, if we use `for a, b in matching` provided we have an edge with three entries:

```
Traceback (most recent call last)
...
ValueError: too many values to unpack (expected 2)
```